osm2streets/src/lib.rs

```diff
@@ -117,10 +119,20 @@ impl StreetNetwork {
         }
     }
 
-    // This always returns roads oriented in clockwise order around the intersection
-    // TODO Consider not cloning. Many callers will have to change
-    pub fn roads_per_intersection(&self, i: osm::NodeID) -> Vec<OriginalRoad> {
-        self.intersections[&i].roads.clone()
+    pub fn remove_intersection(&mut self, id: osm::NodeID) {
+        let i = self.intersections.remove(&id).unwrap();
+        if !i.roads.is_empty() {
+            panic!("Can't remove_intersection({id}), it has roads still connected");
+        }
+    }
+
+    /// Returns roads oriented in clockwise order around the intersection
+    pub fn roads_per_intersection(&self, i: osm::NodeID) -> Vec<&Road> {
```

Comment:
This is an improvement in my eyes :+1:

It makes me wonder (more academically than practically) is it common for rust APIs to return `Iterator<&Road>` in this situation? (I just read through the rust book chapter on iterators, can you tell?)

Reply:
Good question... I haven't read the book in ages and am not sure about best practices actually! If pretty much all callers are just iterating over the list, that'd probably be a better return type, to make things a bit lazier and save an odd manual call to `.into_iter()`.

I've returned something like `&[Road]` before in some cases, but that's only when a `Vec<Road>` is internally stored.

Reply:
Thanks, I'll keep an eye out for APIs returning `Vec` vs `Iterator` to see if a pattern emerges. I'll keep it in the back of my mind as a micro-optimisation I think, avoiding allocating the `Vec` when it's just going to be iterated.